rita_exit/src/database/database_tools.rs

```diff
@@ -348,11 +345,95 @@ pub fn create_or_update_user_record(
 
         let new_ip = get_next_client_ip(conn)?;
 
-        let c = client_to_new_db_client(client, new_ip, user_country);
+        let global_ip = get_client_global_ip(conn)?;
+
+        let c = client_to_new_db_client(client, new_ip, user_country, global_ip);
 
         info!("Inserting new client {}", client.global.wg_public_key);
         diesel::insert_into(clients).values(&c).execute(conn)?;
 
         Ok(c)
     }
 }
+
+/// Out of the subnet provided to the exit, choose an ip that is not in use
+/// Randomly select ips in the subnet until we get a unique one that is not in the database
```

Comment:
1) recursion is probably not the best way to do this
2) we need to make sure subnets don't overlap (so a random IP could be within another clients subnet)